docs/reference/filters.md

```diff
@@ -19,15 +19,31 @@ all: * -> filter1 -> filter2 -> "http://127.0.0.1:1234/";
 ## Template placeholders
 
 Several filters support template placeholders (`${var}`) in string parameters.
-Template placeholder is replaced by the value that is looked up in filter context path parameters.
-Missing template placeholder value interpretation depends on the filter.
+
+Template placeholder is replaced by the value that is looked up in the following sources:
+
+* request headers (if starts with `request.header.` prefix, e.g `${request.header.Content-Type}` is replaced by `Content-Type` request header value)
+* request url query (if starts with `request.url_query.` prefix, e.g `${request.url_query.q}` is replaced by `q` query parameter value)
```

Comment:
i would have named it just simply `request.query`

Reply:
A justification to have `url` in the name was to follow the go API where `query` is a field of `url` which is the field of `request`.

Here I've named it the same as it is named in [lua](https://github.com/zalando/skipper/blob/master/docs/reference/scripts.md#other-request-fields).
There it was named like that `url_<something>` because `url` was already defined as scalar string, so it would not be  possible to use `url.query` (in such case `url` has to be table and `query` would be a key).

So on the one hand it is the same naming as in lua (good) on the other it inherits lua's implementation history (bad).

Reply:
I see, and I missed that before. I leave it to your consideration :slightly_smiling_face:

Reply:
I have changed it to `request.query`:
1. there is no need to follow go API naming
2. makes no sense to inherit lua implementation history
3. there is a `setQuery` filter (not a `setUrlQuery`)

I have also added `request.path` as a complement to the existing `setPath` filter.